### PR TITLE
added functionality to retrieve queue

### PR DESCRIPTION
**Pull request by cfoskin**
Changes:

Refactored code to allow the retrieval of the queue reference to millicore.

Verification:
We need to test new functionality which split out the polling and the triggering of the build from the one build method.
We can now directly call triggerBuild() & pollBuild(). The api of the existing build() method was not changed and should work as previous.

To test: Add this unit test to BuildServiceTest.java : 

``` Java
@Test
    public void shouldTriggerBuildTest() throws Exception {
        DiggerClient diggerClient = DiggerClient.createDefaultWithAuth("JENKINS_URL", "admin", "password");
        Map<String, String> params = new HashMap<String, String>();
        params.put("BRANCH", "master");
        params.put("PLATFORM", "android");
        params.put("FH_CONFIG_CONTENT", "foo");
        params.put("BUILD_CREDENTIAL_ID", "bar");
        params.put("BUILD_CREDENTIAL_ALIAS", "foo");
        params.put("BUILD_CONFIG", "debug");
        String jobName = "JOB_NAME";
        BuildTriggerStatus buildTriggerStatus = diggerClient.triggerBuild(jobName, params);
        diggerClient.pollBuild(jobName, buildTriggerStatus.getQueueReference(), DiggerClient.DEFAULT_BUILD_TIMEOUT, params);
        //BuildTriggerStatus buildTriggerStatus = diggerClient.build(jobName, DiggerClient.DEFAULT_BUILD_TIMEOUT, params);
        diggerClient.cancelBuild(jobName, buildTriggerStatus.getBuildNumber());
    }
```

1. Breakpoint the test at the last line as we want to trigger the build and see it on jenkins & run it in debug
2. On confirming the build is triggered - click through and cancel the build - (Build should cancel successfully)
3. Comment out the  lines which execute diggerClient.triggerBuild & diggerClient.pollBuild & uncomment the diggerClient.build line and run in debug.
4. Ensure the build is triggered on jenkins and then step through to cancel the build (Build should cancel successfully)

This will test that the build method worked as before and we can separately call triggerBuild and pollBuild

### CODE REVIEW

**Comment by cfoskin** [comment]
ping @PhilipGough

**Comment by cfoskin** [comment]
ping @aliok & @PhilipGough

File: src/main/java/org/aerogear/digger/client/model/BuildTriggerStatus.java [hunk 1]
```diff
@@ -47,15 +49,19 @@
         /**
          * The build is stuck on Jenkins queue.
          */
-        STUCK_IN_QUEUE
+        STUCK_IN_QUEUE,
+
```
Comment (philipgough) [inline src/main/java/org/aerogear/digger/client/model/BuildTriggerStatus.java:53]: Maybe add a test description here of when this enum is set

Comment (cfoskin) [reply src/main/java/org/aerogear/digger/client/model/BuildTriggerStatus.java:53]: updated

**Comment by aliok** [comment]
👁

File: src/main/java/org/aerogear/digger/client/DiggerClient.java [hunk 2]
```diff
@@ -250,7 +251,8 @@ public JobWithDetails getJob(String name) throws DiggerClientException {
      */
     public BuildTriggerStatus build(String jobName, long timeout, Map<String, String> params) throws DiggerClientException {
         try {
```
Comment (aliok) [inline src/main/java/org/aerogear/digger/client/DiggerClient.java:253]: maybe we should rename this to sth like `buildAndWait` (I am terrible at naming things)

File: src/main/java/org/aerogear/digger/client/DiggerClient.java [hunk 3]
```diff
@@ -280,6 +282,31 @@ public BuildTriggerStatus build(String jobName, long timeout) throws DiggerClien
     }
 
     /**
+     *
```
Comment (aliok) [inline src/main/java/org/aerogear/digger/client/DiggerClient.java]: A one-liner description of this method would be useful. Sth like 
"Triggers a build for the given job. Unlike {@link build(String, long, Map<String, String>)}, this method returns immediately with the build status, which has a queue reference in it to track the build."

**Review by aliok — CHANGES_REQUESTED** [review]
I think the changes lack some documentation.

Since this is a client, it would be nice to document the public facing parts (especially DiggerClient.java).

File: src/main/java/org/aerogear/digger/client/DiggerClient.java (hunk 2 above)
Comment (philipgough) [reply src/main/java/org/aerogear/digger/client/DiggerClient.java:253]: @aliok I advised this to remain the same as I didn't want to break the existing api, wdyt? More appropriate maybe to concentrate on the naming of the new interfaces?

File: src/main/java/org/aerogear/digger/client/DiggerClient.java (hunk 3 above)
Comment (philipgough) [reply src/main/java/org/aerogear/digger/client/DiggerClient.java]: Agree the doc here should contain info about the purpose of this and how it differs from build

File: src/main/java/org/aerogear/digger/client/DiggerClient.java [hunk 4]
```diff
@@ -280,6 +282,31 @@ public BuildTriggerStatus build(String jobName, long timeout) throws DiggerClien
...
+    /**
+     *
+     * @param jobName name of the job
+     * @param timeout
+     * @param params build parameters to be sent to the Jenkins build
+     * @return The build status
+     * @throws IOException if connection problems occur during connecting to Jenkins
+     * @throws InterruptedException if a problem occurs during sleeping between checks
+     */
+    public BuildTriggerStatus pollBuild(String jobName, QueueReference queueReference, long timeout, Map<String, String> params) throws IOException, InterruptedException{
```
Comment (philipgough) [inline src/main/java/org/aerogear/digger/client/DiggerClient.java:306]: As above a good description here to clarify purpose

**Comment by cfoskin** [comment]
@PhilipGough  @aliok Some docs added as suggested

**Comment by cfoskin** [comment]
Ping @odra

**Comment by odra** [comment]
👍   Verified on OSM4

File: src/main/java/org/aerogear/digger/client/DiggerClient.java (hunk 2 above)
Comment (aliok) [reply src/main/java/org/aerogear/digger/client/DiggerClient.java:253]: ok, makes sense @PhilipGough @cfoskin

**Pull request merged**